.github/workflows/run-e2e-dispatch.yml

```
@@ -0,0 +1,9 @@
+name: E2E staging (dispatched)
+on: workflow_dispatch
+
+jobs:
+  e2e-dispatch:
+    uses: saleor/apps/.github/workflows/e2e.yml@main
```

This would work better for forks and PRs:
```suggestion
    uses: ./.github/workflows/e2e.yml
```